src/adapters/ws/client.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * WS client adapter
+ */
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import ws from 'ws';
+
+
+class WsClientAdapter extends Adapter {
+    private client: ws
+
+    name(): string {
+        return 'WS adapter';
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage) {
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve) => {
+
+            const subscribedChannels = this.getSubscribedChannels();
+            const serverBinding = this.AsyncAPIServer.binding('ws');
+            const securityRequirements = (this.AsyncAPIServer.security() || []).map(sec => {
+                const secName = Object.keys(sec.json())[0]
+                return this.parsedAsyncAPI.components().securityScheme(secName)
+            })
+
+            /**
+             * We do not spin up a server and just create a ws client 
+             * to connect to a existing ws server. 
+             */
+            const url = new URL(this.AsyncAPIServer.url() );
+            console.log(this.serverUrlExpanded);
+            this.client = new ws(url);
+
+            this.client.on('open', () => {
+                this.emit('connect', {name: this.name(), adapter: this, connection: this.client, channels: this.channelNames});
+
+            })
+
+
+            this.client.on('message', (data) => {
+                /**
+                 * For POC I used hard coded chanel name,
+                 * we need to dynamically figure this out. 
+                 */
+                const msg = this._createMessage('listen', data);
+                this.emit('message', msg, this.client);
```

Comment:
@fmvilas I am kinda stuck here. 

Currently, we create adapters according to the number of servers we define in the specification file. This works well in the case of servers as we can parse the URL from the request body to figure out which channel to forward the message but in the case of the client, multiple channels could subscribe to different WebSocket servers or to different paths in the same server (something I am doing in my example). 

Can I somehow reference server in channel, I think this is something v3 will be bringing, but right now how do we determine the channel.

Reply:
Replied in other lines to point you in the right direction.